src/west/main.py

```diff
@@ -65,16 +67,32 @@ def command_handler(command, known_args, unknown_args):
     command.run(known_args, unknown_args)
 
 
-def validate_context(args, unknown):
-    '''Validate the run-time context expected by west.'''
+def ensure_zephyr_base(args, unknown):
+    '''Ensure ZEPHYR_BASE is set, emitting warnings if the user is
+    pointing it somewhere different than what the manifest expects.
+    '''
+
     if args.zephyr_base:
-        os.environ['ZEPHYR_BASE'] = args.zephyr_base
+        zb = args.zephyr_base
+        zb_src = 'the command line'
     else:
-        if 'ZEPHYR_BASE' not in os.environ:
-            log.wrn('--zephyr-base missing and no ZEPHYR_BASE',
-                    'in the environment')
-        else:
-            args.zephyr_base = os.environ['ZEPHYR_BASE']
+        manifest = Manifest.from_file()
+        zb = manifest.zephyr_base
+        zb_src = 'the manifest file {}'.format(manifest.path)
+
+    zb_env = os.environ.get('ZEPHYR_BASE')
```

Comment:
Should you set the source here as well? That way we could later query where the `ZEPHYR_BASE` value came from regardless of where it did.

Reply:
That's a great idea.

Reply:
I think I am ready to approve if you make this change.

Reply:
> I think I am ready to approve if you make this change.

I've included this in the verbose output, let me know if you're looking for something else (like a ZEPHYR_BASE_ORIGIN variable which was also set in the environment).